forc-pkg/src/source/ipfs.rs

```diff
@@ -130,6 +131,24 @@ impl fmt::Display for Pinned {
 }
 
 impl Cid {
+    fn extract_archive<R: std::io::Read>(&self, reader: R, dst: &Path) -> Result<()> {
+        let dst_dir = dst.join(self.0.to_string());
+        std::fs::create_dir_all(&dst_dir)?;
+        let mut archive = Archive::new(reader);
+
+        for entry in archive.entries()? {
+            let mut entry = entry?;
+            let path = entry.path()?;
+            let new_path = dst_dir.join(path);
+            // Create parent directories if they don't exist
```

Comment:
Since the extracted content may not necessarily be trusted; could add security check to prevent path traversal attack(s):
```suggestion
            // Prevent path traversal attacks by checking if path is absolute or contains ..
            if path.is_absolute() || path.components().any(|c| c == Component::ParentDir) {
                return Err(anyhow::anyhow!("Invalid path in archive: contains absolute path or parent directory references"));
            }
```

Reply:
Ahh that is a good catch! Looks like [`Entry::unpack`](https://docs.rs/tar/0.4.38/tar/struct.Entry.html#method.unpack) does not have the built-in checks as `Archive::unpack`.

Instead there is [`Entry::unpack_in`](https://docs.rs/tar/0.4.38/tar/struct.Entry.html#method.unpack_in) which includes this checks to prevent zip slip happening.

Reply:
[5cc8325](https://github.com/FuelLabs/sway/commit/5cc83256f16e1d7d3e02d62d75e5034655e73494) adds a test to demonstrate this is possible. Once CI finalizes and shows that the test is effective, I'll enable the fix

Reply:
[ed29a64](https://github.com/FuelLabs/sway/pull/6902/commits/ed29a64fd7ad0237b9d052f319a4398e58d7099f) fixes this.